### PR TITLE
Various small performance tweaks

### PR DESCRIPTION
**Pull request by toofishes**
Wanted to try my hand at some performance optimization. The biggest hotspots by far are in demodulation and sample conversion, but after that, some of these things fixed here start to show up.

Each commit should be fairly self-contained here in what it fixes, I'm happy to break it apart more or drop certain commits that might need more work.

The basic list of fixes comes down to this:
- Mark functions static when appropriate
- Avoid calls into libc if we can avoid them; tolower() and sprintf() were tackled here

### CODE REVIEW

File: net_io.c [hunk 1]
```diff
@@ -390,7 +390,7 @@ static void modesSendBeastOutput(struct modesMessage *mm) {
 
 static void send_beast_heartbeat(struct net_service *service)
 {
-    static char heartbeat_message[] = { 0x1a, '1', 0, 0, 0, 0, 0, 0, 0, 0, 0 };
+    const char heartbeat_message[] = { 0x1A, '1', 0, 0, 0, 0, 0, 0, 0, 0, 0 };
```
Comment (mutability) [inline net_io.c]: Why?

**Comment by mutability** [comment]
Do you have some profiling results to go with this?

File: net_io.c (hunk 1 above)
Comment (toofishes) [inline net_io.c]: This one doesn't seem like a great idea, now that I'm looking at disassembly. It was mainly for clarity- one could mark these `static const` to better convey intent, but it isn't really necessary. I'll drop this commit.

**Comment by toofishes** [comment]
I'm working on gathering some performance results for you, I'll get them posted here once I have something.

**Comment by Romeo-Golf** [comment]
Were these changes still being considered?

**Pull request closed without merging**